lib/core/src/Cardano/Wallet.hs

```diff
@@ -1431,7 +1422,13 @@ quitStakePool ctx wid pid argGenChange pwd = db & \DBLayer{..} -> do
     walMeta <- mapExceptT atomically $ withExceptT ErrQuitStakePoolNoSuchWallet $
         withNoSuchWallet wid $ readWalletMeta (PrimaryKey wid)
 
-    when (walMeta ^. #delegation /= Delegating pid) $
+    let checkIfJoined (WalletDelegation (Delegating pid') Nothing) =
+            pid' /= pid
+        checkIfJoined (WalletDelegation _ (Just (WalletDelegationNext (Delegating pid') _))) =
+            pid' /= pid
+        checkIfJoined _ = True
+
+    when (checkIfJoined (walMeta ^. #delegation)) $
```

Comment:
Same as for `joinStakePool` :+1: 

Alternatively, these two functions could be moved close to the declaration of the `WalletDelegation` type, as helpers / accessors on the data-type.

Reply:
done